### ISSUE TITLE
gopls compatability

### ISSUE THREAD

**Issue by slnt** [ckaznocha/ide-go#44]
I am using the latest version of:
-   [x] Atom v1.13.1 x64
-   [x] ide-go v0.12.3
-   [x] atom-ide-ui
~-   [ ] go-langserver~

**Operating System**:

macOS 10.14.3

### Expected Behavior

Expected it to JustWork™ when setting the server path to `~/go/bin/gopls`, as it does for when using `bingo`. Since `bingo` is no longer going to be maintained in favour of `gopls` the `ide-go` package should work with the official language server.

### Actual Behavior

A `gopls` process is spawned, but Atom does not communicate with it from what I can tell.

### Steps to Reproduce

`go get -u golang.org/x/tools/cmd/gopls`

Set custom server path to `$(which gopls)`

Reload window, check for process with `ps aux | grep gopls` to see process.

See that atom-ide-ui displays nothing, format on save doesn't work, etc.

**Comment by slnt** [comment]
It seems to work using @saibing's version of `gopls`, however there are a lot of... interesting issues with it it seems. Go module support seems rather lacking at the moment.

I doubt there is anything to do here for this package, except maybe make `gopls` the first class supported citizen rather than `go-languageserver`.

**Comment by slnt** [comment]
Reopening this as `gopls` seems to work as intended with VSCode, however the `ide-go` package does not handle it.

Generally the issues I see are ending up with a lot of "Type X is undeclared" for type outside of whatever file I'm currently working in, even though Type X is in the same package.

Imports are also a bit funky.

**Comment by nwaywood** [comment]
I agree that it would be preferable for this plugin to switch to `gopls` since it is still in active development and supports go modules, as described [here](https://github.com/microsoft/vscode-go#go-language-server)
